lib/screens/welcome_screen.dart

```diff
@@ -1,12 +1,120 @@
+  final TextEditingController _passwordController = TextEditingController();
+  final GoogleSignIn _googleSignIn = GoogleSignIn();
+
+  final _formKey = GlobalKey<FormState>();
+
+  // Toggles the visibility of the password
+  void _togglePasswordVisibility() {
+    setState(() {
+      _obscurePassword = !_obscurePassword;
+    });
+  }
+
+  // Shows error snackbar
+  void _showErrorSnackBar(String message) {
+    ScaffoldMessenger.of(context).showSnackBar(
+      SnackBar(content: Text(message), backgroundColor: Colors.redAccent),
+    );
+  }
+
+  // Login action with simple validation
+  Future<void> _handleLogin() async {
+    if (!_formKey.currentState!.validate()) return;
+
+    setState(() => _isLoading = true);
+
+    await Future.delayed(const Duration(seconds: 2)); // simulate backend call
+
+    setState(() => _isLoading = false);
+
+    Navigator.push(
+      context,
+      MaterialPageRoute(builder: (_) => const HomeScreen(loginMethod: 'Email')),
+    );
+  }
+
+  // Simulate Google Login
+  Future<void> _googleLogin() async {
+    setState(() => _isLoading = true);
+    try {
+      final GoogleSignInAccount? googleUser = await _googleSignIn.signIn();
+      await Future.delayed(const Duration(seconds: 2)); // simulate backend
+
+      setState(() => _isLoading = false);
+
+      if (googleUser != null) {
+        Navigator.push(
+          context,
+          MaterialPageRoute(
+            builder: (_) => const HomeScreen(loginMethod: 'Google'),
+          ),
+        );
+      } else {
+        _showErrorSnackBar('Google sign-in was cancelled.');
+      }
+    } catch (error) {
+      setState(() => _isLoading = false);
+      _showErrorSnackBar('Google Sign-In Error: $error');
+    }
+  }
+
+  // Navigates to the Sign Up form dialog
+  void _navigateToSignUp() {
+    showDialog(
+      context: context,
+      builder:
+          (_) => Dialog(
+            shape: RoundedRectangleBorder(
+              borderRadius: BorderRadius.circular(20),
+            ),
+            elevation: 16,
+            child: Padding(
+              padding: const EdgeInsets.all(20.0),
+              child: SingleChildScrollView(child: SignUpForm()),
+            ),
+          ),
+    );
+  }
+
+  // Forgot Password dialog
+  void _handleForgotPassword() {
+    showDialog(
+      context: context,
+      builder:
+          (_) => Dialog(
+            shape: RoundedRectangleBorder(
+              borderRadius: BorderRadius.circular(20),
+            ),
+            elevation: 16,
+            child: Padding(
+              padding: const EdgeInsets.all(20.0),
+              child: SingleChildScrollView(child: ForgotPasswordForm()),
+            ),
+          ),
+    );
+  }
+
   @override
   Widget build(BuildContext context) {
+    final screenWidth = MediaQuery.of(context).size.width;
+    final isSmallScreen = screenWidth < 600;
+
     return Scaffold(
       body: Container(
         width: double.infinity,
```

Comment:
above code to be implemented here
```suggestion
        width: double.infinity,
        height: screenHeight,
```

Reply:
> Collaborator

solve